frontend/packages/dev-console/src/components/import/ImportForm.tsx

```diff
@@ -171,8 +187,16 @@ const ImportForm: React.FC<ImportFormProps & StateProps> = ({
           });
         }
 
+        const repository = resources.find((resource) => resource.kind === RepositoryModel.kind);
+        if (repository) {
+          history.push(resourcePathFromModel(RepositoryModel, repository.metadata.name, namespace));
+        }
+
         fireTelemetryEvent('Git Import', getTelemetryImport(values));
-        handleRedirect(projectName, perspective, perspectiveExtensions);
+
+        if (selectedStrategy.type !== ImportStrategy.PAC) {
+          handleRedirect(projectName, perspective, perspectiveExtensions);
+        }
```

Comment:
@Lucifergene why this cant be simply like:
`if (selectedStrategy.type === ImportStrategy.PAC) 
   #name and ns are already available in the form
    history.push(resourcePathFromModel(RepositoryModel, name, namespace)); 
  else handleRedirect(projectName, perspective, perspectiveExtensions);
`